src/auto_pickup.h

```diff
@@ -48,13 +48,15 @@ class auto_pickup : public JsonSerializer, public JsonDeserializer
                 ~cRules() {};
         };
 
+        mutable bool ready; //< true if map_items has been populated from vRules
+
         /**
          * The currently-active set of auto-pickup rules, in a form that allows quick
          * lookup. When this is filled (by @ref auto_pickup::create_rules()), every
          * item existing in the game that matches a rule (either white- or blacklist)
          * is added as the key, with RULE_WHITELISTED or RULE_BLACKLISTED as the values.
          */
-        std::unordered_map<std::string, rule_state> map_items;
+        mutable std::unordered_map<std::string, rule_state> map_items;
```

Comment:
See above - this code is ugly, sure, but I am going for the minimum changes to get things working, not a wholesale cleanup.